src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/NewApplicationHandler.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.events;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.PaymentsProcessor;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.casecreation.AutoCaseCreator;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.casecreation.CaseCreationException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.envelopes.model.Envelope;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.processedenvelopes.EnvelopeProcessingResult;
+
+import static java.lang.String.format;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.processedenvelopes.EnvelopeCcdAction.CASE_CREATED;
+import static uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.domains.processedenvelopes.EnvelopeCcdAction.EXCEPTION_RECORD;
+
+/**
+ * Handles envelops of classification NEW_APPLICATION.
+ */
+@Service
+public class NewApplicationHandler {
+
+    public static final int MAX_RETRIES_FOR_POTENTIALLY_RECOVERABLE_FAILURES = 2;
+
+    private final AutoCaseCreator caseCreator;
+    private final PaymentsProcessor paymentsProcessor;
+    private final CreateExceptionRecord exceptionRecordCreator;
+
+    public NewApplicationHandler(
+        AutoCaseCreator caseCreator,
+        PaymentsProcessor paymentsProcessor,
+        CreateExceptionRecord exceptionRecordCreator
+    ) {
+        this.caseCreator = caseCreator;
+        this.paymentsProcessor = paymentsProcessor;
+        this.exceptionRecordCreator = exceptionRecordCreator;
+    }
+
+    public EnvelopeProcessingResult processNewApplication(Envelope envelope, long deliveryCount) {
```

Comment:
is it worth or makes sense to assert classification is definitely NEW_APPLICATION? just in case this handler is misused :shrug:

Reply:
Yeah, it won't hurt adding it